wtxmgr/tx.go

```diff
@@ -1542,6 +1702,133 @@ func (s *Store) UnspentOutputs() ([]*Credit, error) {
 	return credits, err
 }
 
+// outputCreditInfo fetches information about a credit from the database,
+// fills out a credit struct, and returns it.
+func (s *Store) outputCreditInfo(ns walletdb.Bucket, op wire.OutPoint,
+	block *Block) (*Credit, error) {
+	// It has to exists as a credit or an unmined credit.
+	// Look both of these up. If it doesn't, throw an
+	// error. Check unmined first, then mined.
+	var minedCredV []byte
+	unminedCredV := existsRawUnminedCredit(ns,
+		canonicalOutPoint(&op.Hash, op.Index))
+	if unminedCredV == nil {
+		if block != nil {
+			credK := keyCredit(&op.Hash, op.Index, block)
+			minedCredV = existsRawCredit(ns, credK)
+		}
+	}
+	if minedCredV == nil && unminedCredV == nil {
+		errStr := fmt.Errorf("missing utxo %x, %v", op.Hash, op.Index)
+		return nil, storeError(ErrNoExists, "couldn't find relevant credit "+
+			"for unspent output", errStr)
+	}
+
+	// Throw an inconsistency error if we find one.
+	if minedCredV != nil && unminedCredV != nil {
+		errStr := fmt.Errorf("duplicated utxo %x, %v", op.Hash, op.Index)
+		return nil, storeError(ErrNoExists, "credit exists in mined and unmined "+
```

Comment:
Yeah, I was using this incorrectly. I'll add a new error specific to Vs missing in the database.